src/main/java/teammates/logic/api/Logic.java

```diff
@@ -1587,15 +1592,13 @@ public String getFeedbackSessionResultSummaryInSectionAsCsv(String courseId,
      * Preconditions: <br>
      * * All parameters are non-null.
      */
-    public boolean hasStudentSubmittedFeedback(String courseId, String feedbackSessionName, String studentEmail)
+    public boolean hasStudentSubmittedFeedback(FeedbackSessionAttributes fsa, String studentEmail)
             throws InvalidParametersException, EntityDoesNotExistException {
```

Comment:
Yes. Under the old one, there is a repeated query of FeedbackSessionAttributes with the courseId and feedbackSessionName taken from an existing Attributes from a query result, which is redundant.
And it's not being replaced (replacing introduces regression, I've experienced it), I just add another method that takes in the entire Attributes and process it.